xdsl/transforms/canonicalization_patterns/riscv_cf.py

```diff
@@ -0,0 +1,114 @@
+from dataclasses import dataclass
+from typing import Any
+
+from xdsl.dialects.builtin import ModuleOp
+from xdsl.dialects.riscv_cf import BranchOp, ConditionalBranchOperation, JOp
+from xdsl.interpreter import Interpreter
+from xdsl.interpreters import arith, riscv, riscv_debug
+from xdsl.ir import Operation, SSAValue
+from xdsl.pattern_rewriter import PatternRewriter, RewritePattern
+from xdsl.traits import ConstantLike, Pure
+from xdsl.utils.exceptions import InterpretationError
+
+
+class ElideConstantBranches(RewritePattern):
```

Comment:
While this works, I feel like it's adding too much infrastructure for what we need. I also don't feel comfortable adding the interpreter to this lowering. Could you please try with an approach more like in the riscv canonicalization patterns, that checks for the inputs being immediates, and matching on the type of the op?

Reply:
I'll have to agree here; while this is a cool use, going for the usual approach might even make this PR shorter (?)

Reply:
But that doesn't work for the provided filecheck (because the value is "moved"). Implementing this in a manual way would either duplicate a lot of code, or result in inflexible, mess and buggy code imo.